{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BioE 231 Final Project\n",
    "\n",
    "## Team Members:\n",
    "- Jessica Wu\n",
    "- Yuanyuan Fu\n",
    "- Zihui Xu\n",
    "- Xinxin Mo"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Part 1\n",
    "\n",
    "##### 1. Using the SPAdes command line and run it in Screen\n",
    "\n",
    "At the /data/metagenomeWe folder, we find our metagenome, namely 160523Alm_D16-4701_1_sequence.fastq.gz and 160523Alm_D16-4701_2_sequence.fastq.gz. \n",
    "\n",
    "First, we copy the data to our server by using the command line:\n",
    "\n",
    "```sh\n",
    "\n",
    "cp /data/metagenomes/160523Alm_D16-4701_1_sequence.fastq.gz ~/fp/data\n",
    "cp /data/metagenomes/160523Alm_D16-4701_2_sequence.fastq.gz ~/fp/data\n",
    "\n",
    "```\n",
    "\n",
    "Second, we cd to our directory and unzip the .fastq.gz file.\n",
    "```sh\n",
    "gzip -d 160523Alm_D16-4701_1_sequence.fastq.gz\n",
    "gzip -d 160523Alm_D16-4701_2_sequence.fastq.gz\n",
    "\n",
    "```\n",
    "\n",
    "Then we fragmented the first 10000000 lines from the whole data.\n",
    "\n",
    "```sh\n",
    "head -10000000 160523Alm_D16-4701_1_sequence.fastq > cut_reads_1.fastq \n",
    "head -10000000 160523Alm_D16-4701_2_sequence.fastq > cut_reads_2.fastq \n",
    "\n",
    "```\n",
    "\n",
    "Then we used screen and spades the two sequences:\n",
    "\n",
    "```sh\n",
    "\n",
    "spades -1 cut_reads_1.fastq -2 cut_reads_2.fastq -t 2 --meta -o ~/fp/Lab_final_6\n",
    "\n",
    "```\n",
    "\n",
    "annotation of the code above: specify the output directory with --o, the path to the first Illumina read with -1, the path to the second Illumina read with -2\n",
    "\n",
    "After running the SPAdes command, we used the following command line and found the statistical data for the contigs.fasta and scaffolds.fasta files.\n",
    "\n",
    "```\n",
    "\n",
    "be131-03@meowth:~/final_project_x/data/lab_final_6$ assembly-stats contigs.fasta\n",
    "stats for contigs.fasta\n",
    "sum = 177369164, n = 686536, ave = 258.35, largest = 5489\n",
    "N50 = 260, n = 246202\n",
    "N60 = 248, n = 316020\n",
    "N70 = 237, n = 389222\n",
    "N80 = 224, n = 466189\n",
    "N90 = 210, n = 547788\n",
    "N100 = 55, n = 686536\n",
    "N_count = 0\n",
    "Gaps = 0\n",
    "\n",
    "```\n",
    "\n",
    "```\n",
    "\n",
    "be131-03@meowth:~/final_project_x/data/lab_final_6$ assembly-stats scaffolds.fasta\n",
    "stats for scaffolds.fasta\n",
    "sum = 177378834, n = 685578, ave = 258.73, largest = 5489\n",
    "N50 = 260, n = 245275\n",
    "N60 = 248, n = 315079\n",
    "N70 = 237, n = 388271\n",
    "N80 = 224, n = 465232\n",
    "N90 = 210, n = 546829\n",
    "N100 = 55, n = 685578\n",
    "N_count = 9670\n",
    "Gaps = 958\n",
    "\n",
    "```\n",
    "\n",
    "Next we identify the taxon of the given data by typing in the command line and obtaining a gff file:\n",
    "\n",
    "```\n",
    "rna_hmm3.py -i contigs.fasta -o taxon.gff\n",
    "\n",
    "```\n",
    "Once we clean up the file by deleting anything that isn't 16S, we run the command line:\n",
    "\n",
    "```\n",
    "bedtools getfasta -fi contigs.fasta -bed taxon.gff\n",
    "\n",
    "```\n",
    "\n",
    "\n",
    "##### 2. Utilize web services to quantify the abundance of different microbes in the metagenome\n",
    "\n",
    "**2.1 Upload the metagenome to One Codex**\n",
    "\n",
    "We utilized One Codex to analyse the metagenome data. We used the command line as follow to upload:\n",
    "\n",
    "```sh\n",
    "onecodex upload --forward /data/ metagenomes/160523Alm_D16-4701_1_sequence.fastq.gz --reverse /data/metagenomes/160523alm_D16-4701_2_sequence.fasq.gz\n",
    "```\n",
    "\n",
    "**2.2 Analyse the data**\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### For 160523Alm_D16-4701_1_sequence.fastq.gz:\n",
    "\n",
    "One Codex shows that this is a mixed/metagenomic sample 11% of  203,616,980 reads were classified using the One Codex database. \n",
    "\n",
    "The 23.63% of the sample composition are Bacteria. The largest composition group is Armatimonadetes bacterium, which occupies 3.91% of the sample. The second largest group is Acidobacteria bacterium, which occupyies 3.83%. Other components on the list are Terrabacteria group, Candidatus Rokubacteria, Miscellaneous Acidobacteria, Proteobacteria, Streptomyces, and Ktedonobacter racemifer DSM 44963, etc.\n",
    "\n",
    "The graph below shows the the result we got.\n",
    "\n",
    "![1](1.PNG)\n",
    "\n",
    "#### For 160523Alm_D16-4701_2_sequence.fastq.gz:\n",
    "\n",
    "One Codex shows that this is a mixed/metagenomic sample 11% of 203,616,980 reads were classified using the One Codex database.\n",
    "\n",
    "The composition of this sequence is similar to the previous sequence.\n",
    "\n",
    "The graph below shows the the result we got.\n",
    "\n",
    "![2](2.PNG)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**According to the results that we got from One Codex, we can know that the microbes which were living in our soil sample were Bacteria, such as Armatimonadetes bacterium, Acidobacteria bacterium, Terrabacteria group, Candidatus Rokubacteria, Miscellaneous Acidobacteria, Proteobacteria, Streptomyces, and Ktedonobacter racemifer DSM, etc.**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Project ideas\n",
    "1. Figure out region/biome sample was obtained from.\n",
    "We could use the results of our genome assembly and annotation to analyze the most prevalent flora and fauna in the sample. This would allow us to figure out what region/biome these soil samples were obtained from. We could do this by either delving into the One Codex and MG-RAST results to find eukaryotic matches or analyze the most prevalent bacterial species in the sample to see what plant/animal life they are symbiotic with and infer what flora and fauna are found in the region.\n",
    "\n",
    "2. Check for mutations in different types of bacteria. \n",
    "The hypothesis is that due to radioactivity, mutations could arise in bacteria genomes. If we compare the bacteria genomes in the soil sample to reference genomes using BLAST, we can know the extent to which the bacteria has mutated if any. \n",
    "\n",
    "3. Determine whether the soil sample is contaminated or not\n",
    "Based on the abundances of the organisms in the sample, is it possible for us to determine the nature of the soil (whether itâ€™s radioactively contaminated or not)?\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
